### PR TITLE
dev-tool: Add cfssl and cfssljson tool check

### PR DESCRIPTION
**Pull request by sayboras**
VM provisioning with K8S=1 is failing due to missing certs, which is
caused by the unavailability of cfssl and cfssljson tools. This commit
is to add two simple checks in dev-tool to avoid any potential debugging
and troubleshooting time.

```
$ ./contrib/vagrant/scripts/00-create-certs.sh
----- 00-create-certs.sh: main: Generating certificates...
./contrib/vagrant/scripts/cert-gen-helpers.bash: line 50: cfssl: command not found
./contrib/vagrant/scripts/cert-gen-helpers.bash: line 50: cfssljson: command not found
```

Signed-off-by: Tam Mach <sayboras@yahoo.com>

```release-note
dev-tool: Add cfssl and cfssljson tool check
```

### CODE REVIEW

File: tools/dev-doctor/rootcmd.go [hunk 1]
```diff
@@ -147,6 +147,28 @@ func rootCmdRun(cmd *cobra.Command, args []string) {
 			versionRegexp: regexp.MustCompile(`pip (\d+\.\d+\S*)`),
 		},
 		dockerGroupCheck{},
+		&binaryCheck{
+			name:          "golangci-lint",
```
Comment (qmonnet) [inline tools/dev-doctor/rootcmd.go]: The check for `golangci-lint` is already present above, is there any point in re-adding it here or is it a copy-paste leftover?

File: tools/dev-doctor/rootcmd.go [hunk 2]
```diff
@@ -147,6 +147,28 @@ func rootCmdRun(cmd *cobra.Command, args []string) {
...
+			ifNotFound:    checkWarning,
+			versionArgs:   []string{"version"},
+			versionRegexp: regexp.MustCompile(`(\d+\.\d+\S*)`),
+			minVersion:    &semver.Version{Major: 1, Minor: 27, Patch: 0},
+			hint:          "See https://golangci-lint.run/usage/install/#local-installation.",
+		},
+		&binaryCheck{
+			name:          "cfssl",
```
Comment (qmonnet) [inline tools/dev-doctor/rootcmd.go:150]: Should the checks for `cfssl` and `cfssljson` be moved to the list of `binaryCheck`s above, or is there a particular reason to have them below the `dockerGroupCheck{}`?

**Review by qmonnet — CHANGES_REQUESTED** [review]
Good idea, thanks!
Please find a few observations below.

File: tools/dev-doctor/rootcmd.go (hunk 1 above)
Comment (tmcatch) [reply tools/dev-doctor/rootcmd.go:151]: this is clearly the mistake from my side :(, I should have taken a quick look on diff before marking PR ready.

File: tools/dev-doctor/rootcmd.go (hunk 2 above)
Comment (tmcatch) [reply tools/dev-doctor/rootcmd.go:159]: good point, let me move them up. Thanks

**Review by qmonnet — APPROVED** [review]
Looks all good, thank you!

**Review by aditighag — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]

**Comment by pchaigno** [comment]
Not covered by end-to-end tests. Merging.

**Pull request merged**